File: Laboratorio/oct1/part3/Led.py
import time

def encender_led():
    print("[***] LED ENCENDIDO")

def apagar_led():
    print("[   ] LED APAGADO")

def parpadear_led(frecuencia_hz, ciclos):
    if frecuencia_hz <= 0:
        apagar_led()
        return
    medio_periodo_s = 0.5 / frecuencia_hz
    for i in range(ciclos):
        encender_led()
        time.sleep(medio_periodo_s)
        apagar_led()
        time.sleep(medio_periodo_s)

File: Laboratorio/oct1/part1/Led.cpp
#include <iostream>
#include <chrono>
#include <thread>
#include "Led.h"

void encenderLed()
{
    std::cout << "[***] LED ENCENDIDO" << std::endl;
}

void apagarLed()
{
    std::cout << "[   ] LED APAGADO" << std::endl;
}

void parpadearLed(float frecuenciaHz, int ciclos)
{
    if (frecuenciaHz <= 0.0)
    {
        apagarLed();
        return;
    }
    int medioPeriodoMs = 500.0 / frecuenciaHz;
    for (int i = 0; i < ciclos; i++)
    {
        encenderLed();
        std::this_thread::sleep_for(std::chrono::milliseconds(medioPeriodoMs));
        apagarLed();
        std::this_thread::sleep_for(std::chrono::milliseconds(medioPeriodoMs));
    }
}